pyprof/parse/nsight.py

```diff
@@ -72,36 +63,29 @@ def createMarkerTable(self):
         self.db.execute('CREATE INDEX end_index ON marker (end)')
         #self.db.execute('CREATE INDEX id_index ON marker (id)')
 
-    def getCPUInfo(self, corrId):
-        """
-		Given the correlation id, get CPU start, end, thread id, process id.
-		The information can be in the runtime table or the driver table.
-		"""
-
-        #First look in the runtime table
-        #cmd = "select start,end,processId,threadId from {} where correlationId={}".format(self.runtimeT, corrId);
-        cmd = "select start,end,globalTid from {} where correlationId={}".format(self.runtimeT, corrId)
-        result = self.db.select(cmd)
-        assert (len(result) == 1)
-
-        info = result[0]
-        start = info['start']
-        end = info['end']
-        # globalId = f(pid, tid). Call it objId (for legacy).
-        objId = info['globalTid']
-        pid = -1
-        tid = objId & 0x00000000ffffff  # not sure, but appears to be.
-
-        assert (end > start)
-        return [start, end, pid, tid, objId]
+    def encode_object_id(self, info):
+        # Nothing to do for nsight. objId comes out of database
+        assert 'objId' in info
 
     def getKernelInfo(self):
         """
 		Get GPU kernel info
 		"""
-        cmd = "select demangledName as name,correlationId,start,end,deviceId,streamId,gridX,gridY,gridZ,blockX,blockY,blockZ from {}".format(
-            self.kernelT
-        )
+        cmd = ("SELECT "
+              "demangledName as kNameId, "
+              "strings.value as name, "
+              "runtime.start as rStart, "
+              "runtime.end as rEnd, "
+              "runtime.globalTid as objId, "
+              "runtime.globalTid / 0x1000000 % 0x1000000 AS pid, "
+              "runtime.globalTid % 0x1000000 AS tid, "
+              "kernels.globalPid / 0x1000000 % 0x1000000 AS kpid, "
+              "kernels.correlationId,kernels.start,kernels.end,deviceId,streamId,"
+              "gridX,gridY,gridZ,blockX,blockY,blockZ "
+              "FROM {} AS kernels "
+              "JOIN {} AS strings ON (kNameId = strings.Id) "
+              "JOIN {} AS runtime ON (kernels.correlationId = runtime.correlationId AND kpid = pid) "
+              ).format(self.kernelT, self.stringT, self.runtimeT)
```

Comment:
Modern python has moved to fstrings over .format.  See https://realpython.com/python-f-strings/  
Among other reasons, it's faster.

Reply:
Thanks for the info. Now I know what those f strings are!  I kept my changes in the same format as the old code, but I'll use f-strings from now on for any python code I touch.